### PR TITLE
[release-1.11] Cherry-pick E2E tests for broker: TLS key pair rotation #3287

### PR DESCRIPTION
**Pull request by Leo6Leo**
Cherry-pick https://github.com/knative-extensions/eventing-kafka-broker/pull/3287/files

Unblocked when https://github.com/knative/hack/pull/350 is merged.

### CODE REVIEW

**Comment by Cali0707** [comment]
@Leo6Leo can you look into why the kafka-source-dispatcher pod is not coming live?

File: test/e2e-common.sh [hunk 1]
```diff
@@ -91,15 +91,27 @@ function knative_eventing() {
...
+    kubectl get issuers --all-namespaces
+    kubectl describe issuers selfsigned-ca-issuer -n=knative-eventing
+    kubectl get issuers selfsigned-ca-issuer -n=knative-eventing -o yaml
+
+
+
+#
+#  else
+#    echo ">> Install Knative Eventing from ${KNATIVE_EVENTING_RELEASE}"
+#    kubectl apply -f "${KNATIVE_EVENTING_RELEASE}"
```
Comment (pierDipi) [inline test/e2e-common.sh:113]: Maybe that's not enough to include TLS resources?

File: data-plane/pom.xml [hunk 2]
```diff
@@ -61,7 +61,7 @@
     <kafka.version>3.2.3</kafka.version>
     <debezium.version>1.9.6.Final</debezium.version>
     <jib.version>3.3.2</jib.version>
-    <quarkus.version>3.2.7.Final</quarkus.version>
+    <quarkus.version>3.2.0.Final</quarkus.version>
```
Comment (pierDipi) [inline data-plane/pom.xml:64]: why are we downgrading components?

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**